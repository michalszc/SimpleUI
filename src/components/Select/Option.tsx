import React, { FC, MouseEventHandler } from "react";
import styled, { css } from "styled-components";
import Styles, { StyleProps } from "../../utils/styles";
import { OptionValue } from "../../utils/types/option";

export interface OptionProps extends StyleProps {
    /**
     * Value of the option
     */
    value: OptionValue;
   /**
    * If true -> option has been selected
    */
    isChecked?: boolean;
    /**
     * If true -> option can't be selected
     */
    isDisable?: boolean;

    onClick?: MouseEventHandler<HTMLAnchorElement>;
}

const StyledOptionContainer = styled.a<OptionProps>`
    font-family: sans-serif;
    color: #555555;
    padding: 10px 5px;
    width: 100%;
    cursor: pointer;
    transition: transform .2s ease-in-out;

    &:hover {
        transform: scale(1.02);
    }

    ${({ isChecked }) => isChecked && css`
        font-size: 1.05em;
        color: #1d99ff;

    `}

`;

const Option: FC<OptionProps> = ({ onClick, value, isChecked, ...props }) => {
    return(
        <Styles {...props}>
            <StyledOptionContainer 
                value={value}
                isChecked={isChecked} 
                key={value.index}
                onClick={onClick}>
                {value.value}
            </StyledOptionContainer>
        </Styles>
    );
};

export default Option;
