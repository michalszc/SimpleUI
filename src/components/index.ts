export * from "./Button";
export * from "./ColorPicker";
export * from './Switch';
export * from './FormLabel';