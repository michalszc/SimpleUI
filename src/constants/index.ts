export { 
    default as Colors,
    red,
    pink,
    purple,
    deepPurple,
    indigo,
    blue,
    lightBlue,
    cyan,
    teal,
    green,
    lightGreen,
    lime,
    yellow,
    amber,
    orange,
    deepOrange,
    brown,
    grey,
    blueGrey,
    black,
    white,
    transparent
} from './Colors';
export type { Color, IColors } from './Colors';
export { default as Palette } from './Palette';
export type { IPalette, PaletteColor } from './Palette';
export { default as Typography } from './Typography';
export type { ITypography, TypographyVariant } from './Typography';
export { default as Breakpoints } from './Breakpoints';
export type { IBreakpoints } from './Breakpoints';
export type { AbsoluteUnits, RelativeUnits, Units } from './units';
export { default as Sizes } from './Sizes';
export type { ISizes } from './Sizes';
