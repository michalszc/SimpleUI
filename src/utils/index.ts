export * from "./types";
export type { StyleProps } from "./styles";
export * from './theme';
export * from './convertColors';