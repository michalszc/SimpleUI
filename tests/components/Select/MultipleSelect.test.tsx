import React from "react";
import { render } from "@testing-library/react";
import { MultipleSelect } from "../../../src";
import Option from "../../../src/components/Select/Option";

describe("Multiple Select", () => {
    const data =  [
        'First option', 
        'Second option',
        'Third option',
    ];
    const mockFn = jest.fn();
    const color = "#ff0000";

    test("renders the Multiple Select component", () => {
        const container = render(
            <MultipleSelect placeholder="Search value" onChange={(v) => mockFn(v)}>
                <Option value={data[0]}/>
                <Option value={data[1]}/>
                <Option value={data[2]}/>
            </MultipleSelect>
        );

        expect(container).toMatchSnapshot();
    });

    test("renders the Multiple Select component with changed border-color", () => {
        const container = render(
            <MultipleSelect borderColor={color} placeholder="Search value" onChange={(v) => mockFn(v)}>
                <Option value={data[0]}/>
                <Option value={data[1]}/>
                <Option value={data[2]}/>
            </MultipleSelect>
        );
        expect(container).toMatchSnapshot();
    });
});
