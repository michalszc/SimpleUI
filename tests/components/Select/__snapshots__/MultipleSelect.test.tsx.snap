// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders the Multiple Select component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-pyfCe ipOEpi"
      >
        <div
          class="sc-jrcTuL igLDLY simpleui-select"
        >
          <div
            class="sc-kDvujY iOuutZ simpleui-select-inputcontainer"
          >
            <input
              class="sc-ipEyDJ cVmQtJ simpleui-select-input"
              placeholder="Search value"
              value=""
            />
            <a
              class="sc-csuSiG fRNizA"
            >
              <svg
                fill="currentColor"
                height="1.5rem"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 13.172l4.95-4.95 1.414 1.414L12 16 5.636 9.636 7.05 8.222z"
                  />
                </g>
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-pyfCe ipOEpi"
    >
      <div
        class="sc-jrcTuL igLDLY simpleui-select"
      >
        <div
          class="sc-kDvujY iOuutZ simpleui-select-inputcontainer"
        >
          <input
            class="sc-ipEyDJ cVmQtJ simpleui-select-input"
            placeholder="Search value"
            value=""
          />
          <a
            class="sc-csuSiG fRNizA"
          >
            <svg
              fill="currentColor"
              height="1.5rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 13.172l4.95-4.95 1.414 1.414L12 16 5.636 9.636 7.05 8.222z"
                />
              </g>
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Select renders the Multiple Select component with changed border-color 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-pyfCe ipOEpi"
      >
        <div
          class="sc-jrcTuL igLDLY simpleui-select"
          style="border-color: #ff0000;"
        >
          <div
            class="sc-kDvujY iOuutZ simpleui-select-inputcontainer"
          >
            <input
              class="sc-ipEyDJ cVmQtJ simpleui-select-input"
              placeholder="Search value"
              value=""
            />
            <a
              class="sc-csuSiG fRNizA"
            >
              <svg
                fill="currentColor"
                height="1.5rem"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 13.172l4.95-4.95 1.414 1.414L12 16 5.636 9.636 7.05 8.222z"
                  />
                </g>
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-pyfCe ipOEpi"
    >
      <div
        class="sc-jrcTuL igLDLY simpleui-select"
        style="border-color: #ff0000;"
      >
        <div
          class="sc-kDvujY iOuutZ simpleui-select-inputcontainer"
        >
          <input
            class="sc-ipEyDJ cVmQtJ simpleui-select-input"
            placeholder="Search value"
            value=""
          />
          <a
            class="sc-csuSiG fRNizA"
          >
            <svg
              fill="currentColor"
              height="1.5rem"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 13.172l4.95-4.95 1.414 1.414L12 16 5.636 9.636 7.05 8.222z"
                />
              </g>
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
